packages/types/src/discord/components.ts

```diff
@@ -0,0 +1,344 @@
+/** https://discord.com/developers/docs/components/reference#button-button-styles */
+export enum ButtonStyles {
+  /** A blurple button */
+  Primary = 1,
+  /** A grey button */
+  Secondary,
+  /** A green button */
+  Success,
+  /** A red button */
+  Danger,
+  /** A button that navigates to a URL */
+  Link,
+  /** A blurple button to show a Premium item in the shop */
+  Premium,
+}
+
+/**
+ * https://discord.com/developers/docs/components/reference#string-select
+ * https://discord.com/developers/docs/components/reference#user-select
+ * https://discord.com/developers/docs/components/reference#role-select
+ * https://discord.com/developers/docs/components/reference#mentionable-select
+ * https://discord.com/developers/docs/components/reference#channel-select
+ */
+export interface DiscordSelectMenuComponent extends DiscordBaseComponent {
+  type:
+    | MessageComponentTypes.SelectMenu
+    | MessageComponentTypes.SelectMenuUsers
+    | MessageComponentTypes.SelectMenuRoles
+    | MessageComponentTypes.SelectMenuUsersAndRoles
+    | MessageComponentTypes.SelectMenuChannels
+
+  /** A custom identifier for this component. Maximum 100 characters. */
+  custom_id: string
+  /** Specified choices in a select menu; Maximum of 25 items. */
+  options?: DiscordSelectOption[]
+  /** A custom placeholder text if nothing is selected. Maximum 150 characters. */
+  placeholder?: string
+  /** The minimum number of items that must be selected. Default 1. Between 1-25. */
+  min_values?: number
+  /** The maximum number of items that can be selected. Default 1. Between 1-25. */
+  max_values?: number
+  /**
+   * Whether select menu is disabled
+   *
+   * @default false
+   */
+  disabled?: boolean
+  /**
+   * List of default values for auto-populated select menu components
+   *
+   * @remarks
+   * The number of default values must be in the range defined by min_values and max_values
+   */
+  default_values?: DiscordSelectMenuDefaultValue[]
+  /** List of channel types to include in a channel select menu options list */
+  channel_types?: ChannelTypes[]
+}
+
+/** https://discord.com/developers/docs/components/reference#string-select-select-option-structure */
+export interface DiscordSelectOption {
+  /** The user-facing name of the option. Maximum 25 characters. */
+  label: string
+  /** The dev-defined value of the option. Maximum 100 characters. */
+  value: string
+  /** An additional description of the option. Maximum 50 characters. */
+  description?: string
+  /** The id, name, and animated properties of an emoji. */
+  emoji?: {
+    /** Emoji id */
+    id?: string
+    /** Emoji name */
+    name?: string
+    /** Whether this emoji is animated */
+    animated?: boolean
+  }
+  /** Will render this option as already-selected by default. */
+  default?: boolean
+}
+
+/** https://discord.com/developers/docs/components/reference#text-input-text-input-structure */
+export interface DiscordInputTextComponent extends DiscordBaseComponent {
```

Comment:
```suggestion
export interface DiscordTextInputComponent extends DiscordBaseComponent {
```
This would be a type breaking change, but it took me a minute to search the codebase not knowing it's InputText, like why did you change the name Discord